protocols/v2/subprotocols/template-distribution/src/new_template.rs

```diff
@@ -10,51 +10,56 @@ use core::convert::TryInto;
 #[cfg(all(feature = "with_serde", not(feature = "no_std")))]
 use std::convert::TryInto;
 
-/// ## NewTemplate (Server -> Client)
-/// The primary template-providing function. Note that the coinbase_tx_outputs bytes will appear
-/// as is at the end of the coinbase transaction.
+/// Message used by an upstream to provide a new template for downstream to mine on.
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
 pub struct NewTemplate<'decoder> {
-    /// Server’s identification of the template. Strictly increasing, the
-    /// current UNIX time may be used in place of an ID.
+    /// Upstream’s identification of the template.
+    ///
+    /// Strictly increasing, the current UNIX time may be used in place of an ID.
     pub template_id: u64,
-    /// True if the template is intended for future [`crate::SetNewPrevHash`]
-    /// message sent on the channel. If False, the job relates to the last
-    /// sent [`crate::SetNewPrevHash`] message on the channel and the miner
-    /// should start to work on the job immediately.
+    /// If `True`, the template is intended for future [`crate::SetNewPrevHash`] message sent on
+    /// the channel.
+    ///
+    /// If `False`, the job relates to the last sent [`crate::SetNewPrevHash`] message on the
+    /// channel and the miner should start to work on the job immediately.
     pub future_template: bool,
-    /// Valid header version field that reflects the current network
-    /// consensus. The general purpose bits (as specified in [BIP320]) can
-    /// be freely manipulated by the downstream node. The downstream
-    /// node MUST NOT rely on the upstream node to set the BIP320 bits
-    /// to any particular value.
+    /// Valid header version field that reflects the current network consensus.
+    ///
+    /// The general purpose bits, as specified in
+    /// [BIP320](https://github.com/bitcoin/bips/blob/master/bip-0320.mediawiki), can be freely
+    /// manipulated by the downstream node.
+    ///
+    /// The downstream node **must not** rely on the upstream node to set the
+    /// [BIP320](https://github.com/bitcoin/bips/blob/master/bip-0320.mediawiki) bits to any
+    /// particular value.
     pub version: u32,
-    /// The coinbase transaction nVersion field.
+    /// The coinbase transaction `nVersion` field.
     pub coinbase_tx_version: u32,
-    /// Up to 8 bytes (not including the length byte) which are to be placed
-    /// at the beginning of the coinbase field in the coinbase transaction.
+    /// Up to 8 bytes (not including the length byte) which are to be placed at the beginning of
+    /// the coinbase field in the coinbase transaction.
```

Comment:
This is wrong (cc @plebhash), since they are the bip34 bytes, we should put a proper description for it.